### ISSUE TITLE
Jackett keeps reloading with v0.10.1.0 on MacOS

### ISSUE THREAD

**Issue by dkids** [Jackett/Jackett#3501]
I had a problem with the automatic upgrade and untarred the v0.10.1.0 version on top of the previous version.   This is MacOS 10.13.6

018-07-31 14:31:07.7366 Info Loading Cardigann definition /Applications/Jackett/Definitions/ztracker.yml 
2018-07-31 14:31:07.7509 Info Adding aggregate indexer 
2018-07-31 14:31:12.2100 Info Starting Jackett v0.10.1.0 
2018-07-31 14:31:14.1551 Info Environment version: 4.0.30319.42000 (/Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/4.5) 
2018-07-31 14:31:14.1563 Info OS version: Unix 17.7.0.0 (64bit OS) (64bit process) 
2018-07-31 14:31:14.1571 Info ThreadPool MaxThreads: 400 workerThreads, 200 completionPortThreads 
2018-07-31 14:31:14.1571 Info App config/log directory: /Users/kids/.config/Jackett 
2018-07-31 14:31:14.1586 Info mono version: 5.10.1.47 (2017-12/8eb8f7d5e74 Fri Apr 13 20:18:12 EDT 2018) 
2018-07-31 14:31:14.3312 Info TrustedRootCertificates count: 143 
2018-07-31 14:31:14.3397 Info Using HTTP Client: HttpWebClient 
2018-07-31 14:31:15.0816 Info Loading Cardigann definitions from: /Users/kids/.config/cardigann/definitions/, /etc/xdg/cardigan/definitions/, /Applications/Jackett/Definitions 
2018-07-31 14:31:15.1226 Info Loading Cardigann definition /Applications/Jackett/Definitions/1337x.yml 
2018-07-31 14:31:15.2552 Info Loading Cardigann definition /Applications/Jackett/Definitions/2fast4you.yml
[...]
2018-07-31 14:31:18.1598 Info Adding aggregate indexer 
2018-07-31 14:31:22.7858 Info Starting Jackett v0.10.1.0 
2018-07-31 14:31:24.7884 Info Environment version: 4.0.30319.42000 (/Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/4.5) 
2018-07-31 14:31:24.7894 Info OS version: Unix 17.7.0.0 (64bit OS) (64bit process) 
2018-07-31 14:31:24.7902 Info ThreadPool MaxThreads: 400 workerThreads, 200 completionPortThreads 
2018-07-31 14:31:24.7902 Info App config/log directory: /Users/kids/.config/Jackett 
2018-07-31 14:31:24.7915 Info mono version: 5.10.1.47 (2017-12/8eb8f7d5e74 Fri Apr 13 20:18:12 EDT 2018) 
2018-07-31 14:31:24.9010 Info TrustedRootCertificates count: 143 
2018-07-31 14:31:24.9101 Info Using HTTP Client: HttpWebClient 
2018-07-31 14:31:25.5310 Info Loading Cardigann definitions from: /Users/kids/.config/cardigann/definitions/, /etc/xdg/cardigan/definitions/, /Applications/Jackett/Definitions 
2018-07-31 14:31:25.5702 Info Loading Cardigann definition /Applications/Jackett/Definitions/1337x.yml 
2018-07-31 14:31:25.6939 Info Loading Cardigann definition /Applications/Jackett/Definitions/2fast4you.yml 
[...]
2018-07-31 14:31:39.6128 Info Adding aggregate indexer 
2018-07-31 14:31:44.4578 Info Starting Jackett v0.10.1.0 
2018-07-31 14:31:46.6696 Info Environment version: 4.0.30319.42000 (/Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/4.5) 
2018-07-31 14:31:46.6708 Info OS version: Unix 17.7.0.0 (64bit OS) (64bit process) 
2018-07-31 14:31:46.6716 Info ThreadPool MaxThreads: 400 workerThreads, 200 completionPortThreads 
2018-07-31 14:31:46.6722 Info App config/log directory: /Users/kids/.config/Jackett 
2018-07-31 14:31:46.6731 Info mono version: 5.10.1.47 (2017-12/8eb8f7d5e74 Fri Apr 13 20:18:12 EDT 2018) 
2018-07-31 14:31:46.8873 Info TrustedRootCertificates count: 143 
2018-07-31 14:31:46.8909 Info Using HTTP Client: HttpWebClient 
2018-07-31 14:31:47.5312 Info Loading Cardigann definitions from: /Users/kids/.config/cardigann/definitions/, /etc/xdg/cardigan/definitions/, /Applications/Jackett/Definitions 
[...]
loads again

**Comment by flightlevel** [comment]
From the terminal, can you please run `mono --debug JackettConsole.exe -t -l` and provide the full output

**Comment by dkids** [comment]
$ mono --debug Jackett/JackettConsole.exe -t -l 
08-01 08:59:13 Info Starting Jackett v0.10.1.0 
08-01 08:59:13 Info Logging enabled. 
08-01 08:59:13 Info Tracing enabled. 
08-01 08:59:13 Debug Creating web host... 

Eventually I pushed Ctrl-C

$ date
Wed Aug  1 09:18:30 PDT 2018

So for 19 minutes that's all I had.

I don't see a way of increasing the debug level.   I'm looking into the man pages but probably can't do any more debugging until Sunday evening.   Sorry.

$ mono --help-debug
Debugging options:
   --debug[=OPTIONS]     Enable debugging support, optional OPTIONS is a comma
                         separated list of options

OPTIONS is composed of:
    casts                Enable more detailed InvalidCastException messages.
    mdb-optimizations    Disable some JIT optimizations which are normally
                         disabled when running inside the debugger.
                         This is useful if you plan to attach to the running
                         process with the debugger.

**Comment by doc1011** [comment]
Same issue as above since autoupdate v0.10.1.0 on macOS 10.13.6

Previously working without issue, no other changes on server. 

Hangs on mono --debug JackettConsole.exe -t -l

**Comment by flightlevel** [comment]
Tested on OSX with Mono 5.12 and its working well

@doc1011 What version of Mono are you using?

@dkids Could you please try upgrading to Mono 5.12 and see if that makes any difference

@dkids @doc1011 
If it's still occurring
Open a terminal, cd into the directory that contains the Jackett application files
Run `export MONO_LOG_LEVEL=debug`
Then run `mono --debug JackettConsole.exe > output.txt`
Wait a minute, try opening Jackett in browser and then CTRL + C at terminal and provide the output.txt file

**Comment by doc1011** [comment]
@flightlevel Cheers mate - Mono version 5.10.1.47

attached is output.txt

[output.txt](https://github.com/Jackett/Jackett/files/2254970/output.txt)

**Comment by flightlevel** [comment]
Thanks @doc1011 
`08-02 20:49:30 Fatal Application startup exception Autofac.Core.DependencyResolutionException: An error occurred during the activation of a particular registration.`
Suspect this is a bug in Mono 5.10 for OSX
Can you please try on Mono 5.12
https://www.mono-project.com/download/stable/#download-mac

**Comment by doc1011** [comment]
@flightlevel  

Updated mono to 5.12. Sonarr and Radarr still working, mono --version showing 5.12, but same issue with Jackett. I've run the debug again and have attached the output.

[outputv2.txt](https://github.com/Jackett/Jackett/files/2258602/outputv2.txt)

**Comment by flightlevel** [comment]
@doc1011 That's something!
Frustrating I can't reproduce this on OSX
Some changes have been made in the latest pre-release v0.10.5, can you please give that a try?

**Comment by doc1011** [comment]
@flightlevel the pre-release works - all systems go! 

Cheers mate, your support has been much appreciated

**Comment by flightlevel** [comment]
Thanks @doc1011 nice to hear

@dkids Can you please try the latest pre-release to confirm its now resolved the issue for you as well

**Comment by dkids** [comment]
Just got home from family reunion.   Not working for me with Mono 5.10.   One thing is that I haven't used a password for Jackett.   I try pressing the login button with a blank password and nothing is happening.   Attached is the log.

I'm willing to upgrade to Mono 5.12 since I don't use Visual Source Studio.   I'll try that and report how it goes.

[jackett.txt](https://github.com/Jackett/Jackett/files/2260702/jackett.txt)

**Comment by flightlevel** [comment]
@dkids That log looks a lot more promising
Can you please try a different browser/incognito mode/clear cookies/turn off Ad-block and see if that makes a difference?

**Comment by dkids** [comment]
I'm normally using Firefox with uBlock, DuckDuckGo privacy essentials, First Party Isolation, etc.

I tried using stock Safari and this is the result:
[jackett.txt](https://github.com/Jackett/Jackett/files/2262963/jackett.txt)

It's longer than the other two files I've uploaded, like twice as long.   Sorry if that's a problem.

**Comment by flightlevel** [comment]
@dkids Did it work with Safari? That log looks good to me, I can see you are getting heaps of results back from different indexers

**Comment by dkids** [comment]
Thanks for following up.   I can't login with safari, either.   Does this help?

2018-08-10 21:05:55.7260 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/UI/Dashboard
2018-08-10 21:05:55.7272 Info Route matched with {action = "Dashboard", controller = "WebUI"}. Executing action Jackett.Server.Controllers.WebUIController.Dashboard (JackettConsole)
2018-08-10 21:05:55.7366 Info Authorization failed.
2018-08-10 21:05:55.7366 Info Authorization failed for the request at filter 'Microsoft.AspNetCore.Mvc.Authorization.AuthorizeFilter'.
2018-08-10 21:05:55.7378 Info Executing ChallengeResult with authentication schemes ().
2018-08-10 21:05:55.7649 Info AuthenticationScheme: Cookies was challenged.
2018-08-10 21:05:55.7653 Info Executed action Jackett.Server.Controllers.WebUIController.Dashboard (JackettConsole) in 29.0862ms
2018-08-10 21:05:55.7661 Info Request finished in 40.1443ms 302
2018-08-10 21:05:55.7894 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/UI/Login?ReturnUrl=%2FUI%2FDashboard
2018-08-10 21:05:55.8005 Info Route matched with {action = "Login", controller = "WebUI"}. Executing action Jackett.Server.Controllers.WebUIController.Login (JackettConsole)
2018-08-10 21:05:55.8013 Info Executing action method Jackett.Server.Controllers.WebUIController.Login (JackettConsole) - Validation state: Valid
2018-08-10 21:05:55.8025 Info Executed action method Jackett.Server.Controllers.WebUIController.Login (JackettConsole), returned result Microsoft.AspNetCore.Mvc.PhysicalFileResult in 0.1317ms.
2018-08-10 21:05:55.8219 Info Executing Microsoft.AspNetCore.Mvc.PhysicalFileResult, sending file '/Applications/Jackett/Content/login.html' with download name '' ...
2018-08-10 21:05:55.8229 Info Executed action Jackett.Server.Controllers.WebUIController.Login (JackettConsole) in 21.9934ms
2018-08-10 21:05:55.8247 Info Request finished in 35.1527ms 200 text/html
2018-08-10 21:05:56.3469 Info Found 0 releases from World Wide Torrents for: monstress 003
2018-08-10 21:05:56.3475 Info Executed action method Jackett.Server.Controllers.ResultsController.Torznab (JackettConsole), returned result Microsoft.AspNetCore.Mvc.ContentResult in 2278.9615ms.
2018-08-10 21:05:56.3591 Info Executing ContentResult with HTTP Response ContentType of application/rss+xml; charset=utf-8
2018-08-10 21:05:56.3601 Info Executed action Jackett.Server.Controllers.ResultsController.Torznab (JackettConsole) in 2306.5419ms
2018-08-10 21:05:56.3601 Info Request finished in 3739.2625ms 200 application/rss+xml; charset=utf-8
2018-08-10 21:05:56.3744 Info Request starting HTTP/1.1 GET http://127.0.0.1:9117/api/v2.0/indexers/btdb/results/torznab?t=search&q=monstress%20003&cat=100001&apikey=hzapwbdy71nvworkn9xotkttpw3vz6vj
2018-08-10 21:05:56.3751 Info Route matched with {action = "Torznab", controller = "Results"}. Executing action Jackett.Server.Controllers.ResultsController.Torznab (JackettConsole)
2018-08-10 21:05:56.3902 Info Executing action method Jackett.Server.Controllers.ResultsController.Torznab (JackettConsole) with arguments (Jackett.Common.Models.DTO.TorznabRequest) - Validation state: Valid

**Comment by flightlevel** [comment]
Can you attempt to go to http://127.0.0.1:9117 in Chrome incognito mode. Then press F12, Network, tick disable cache, then press CTRL + F5 and post a screenshot of the results

![image](https://user-images.githubusercontent.com/9430010/43999492-8abff2a8-9e50-11e8-81c9-aa1ab5972595.png)

**Comment by kaso17** [comment]
Starting with 0.10.12 I got a login redirect loop (due to the cookie changes) on my mono jackett instance.
Clearing the old cookie fixed it for me. If it's working in an incognito tab you might want to try that too.

**Comment by dkids** [comment]
Sorry for the delayed response.   I'm having trouble providing a screenshot because I'm running a headless Mac Mini and the screenshots are all blank.   

I've tried Firefox (private mode), Chrome (incognito mode), and Safari.  I've wiped all cookies in both Chrome and Safari.    Firefox has a screenshot option when you push the three dots in the nav bar but unfortunately it doesn't work on the developer tools.

While I was writing this, Sonarr downloaded a TV episode so I'm wondering if the issue is running headless.   I'm going to go connect a monitor, keyboard and mouse to the Mac Mini and see if it makes a difference.

I'm doing the following:

1) erasing /Applications/Jackett/
2) launchctl unload ~/Library/LaunchAgents/org.user.Jackett.plist
3) using tar xvf with the latest version (0.10.12)
4) launchctl load ~/Library/LaunchAgents/org.user.Jackett.plist

 In all 3 of them I get something like this

(Chrome, after deleting cookies and history, incognito mode)

Dashboard ( Request URL: http://127.0.0.1:9117/UI/Dashboard
                      Request Method: POST
                      Status Code: 302 Found)
Dashboard ( Request URL: http://127.0.0.1:9117/UI/Dashboard
                      Request Method: POST
                      Status Code: 302 Found)
(The response headers are slightly different; the second one includes:

Location: http://127.0.0.1:9117/UI/Login?ReturnUrl=%2FUI%2FDashboard
Server: Kestrel
)

Login?ReturnUrl=%2FUI%2FDashboard
jquery.min.js
jquery.DataTablesmin.js
handlebars.min.js
moment.min.js
handlebarsmoment.js
bootstrap.min.js
bootstrap-notify.js
bootstrap/bootstrap.min.css
animate.css
custom.css
jacket_medium.png
binding_dark.png
favicon.ico

I notice I don't get the same items that you are.   Ctrl+F5 is not seeming to be transmitted over screen-sharing.   Hopefully better luck with actual devices.

**Comment by dkids** [comment]
Sonarr has downloaded a second TV episode.   I'm prepared to say that this is an issue with just my system if it's working OK for other people.   And Sonarr just did a third.   Search in Radarr works.

This is with physical devices hooked up.

Chrome
![screen shot 2018-08-12 at 8 54 16 pm](https://user-images.githubusercontent.com/30055574/44012157-fc4f3cd2-9e71-11e8-93d5-aac05c55c29d.png)  

Ctrl+F5 is not working to refresh.   I googled it and Command-R is the Mac equivalent.   I get the same results.

Safari
![screen shot 2018-08-12 at 8 58 05 pm](https://user-images.githubusercontent.com/30055574/44012286-e0e56f74-9e72-11e8-88f6-de88cb30d02c.png)

Firefox, private browsing
![screen shot 2018-08-12 at 8 59 13 pm](https://user-images.githubusercontent.com/30055574/44012297-ea2c8612-9e72-11e8-96b4-ade3d94d8f81.png)

**Comment by flightlevel** [comment]
Okay, so everything works fine with peripherals plugged in, its just headless that is the issue?

**Comment by dkids** [comment]
I can't use (meaning log into) Jackett with any of the 3 browsers, headless or not.   I've tried clearing history, cookies, using private browsing, incognito mode, or private window (Safari's term).

However, it doesn't seem to be affecting Sonarr or Radarr, which are working fine with 0.10.12.

It just seems to be affecting me.   I really have no idea what could be different about my setup but there's a new version of Mac OS around the corner, Firefox and Chrome and Jackett are continually upgrading and I can keep digging into the situation.

I can, if you want, install 0.10.12 on a different machine or a different user on the same machine.   But you've got reports that it's working on other macs, maybe with clearing cookies.  

I just installed Mono 5.14.   That didn't make a difference.

I'm going to try installing 0.9.41, adding a password, then upgrading to 0.10.12.   

OK, that fixed it!   Works in all 3 browsers.   

Not sure if that can be fixed in code or maybe you require people setting a password.   My guess is that Radarr and Sonarr worked fine because they used the API Key, not the login process.

Thanks so much for your patience as I worked that out.
